### PR TITLE
Sake contract & mint contract

### PR DESCRIPTION
**Pull request by GalaxySciTech**

### CODE REVIEW

File: endpoint/contracts/examples/treasury/MintTreasury.sol [hunk 1]
```diff
@@ -24,17 +26,40 @@ contract MintTreasury {
...
+        uint256 amount,
+        address recv
+    ) public {
+        require(
+            token == treasuryMapping[rid][originalToken],
+            "token not match"
+        );
         TreasuryToken(token).burnFrom(msg.sender, amount);
+        bytes memory data = abi.encodeWithSelector(
+            bytes4(keccak256("unlock(address,uint256,address)")),
```
Comment (wgr523) [inline endpoint/contracts/examples/treasury/MintTreasury.sol]: question just for clarity: which contract's function `unlock` this is calling?

File: endpoint/contracts/examples/treasury/LockTreasury.sol [hunk 2]
```diff
@@ -35,17 +35,35 @@ contract LockTreasury {
         string memory name = IERC20Metadata(token).name();
         string memory symbol = IERC20Metadata(token).symbol();
         bytes memory data = abi.encodeWithSelector(
-            bytes4(keccak256("mint(address,string,string,address,uint256)")),
+            bytes4(
+                keccak256("mint(address,string,string,address,uint256,uint256)")
```
Comment (wgr523) [inline endpoint/contracts/examples/treasury/LockTreasury.sol:39]: question just for clarity: which contract's function `mint` this is calling?

File: endpoint/contracts/examples/treasury/LockTreasury.sol [hunk 3]
```diff
@@ -35,17 +35,35 @@ contract LockTreasury {
...
-        IEndpoint(_endpoint).send(_rid, _rua, data);
+        IEndpoint(_endpoint).send(rid, rua, data);
+    }
+
+    function unlock(
+        address token,
+        uint256 amount,
+        address recv
+    ) external onlyEndpoint {
+        IERC20Metadata(token).safeTransfer(recv, amount);
```
Comment (wgr523) [inline endpoint/contracts/examples/treasury/LockTreasury.sol:56]: why `safeTransfer` can be called, but others need `endpoint.send`?

**Review by wgr523 — COMMENTED** [review]
For LockTreasury:
- Why inside `lock` we send to `mint` function?
- Why inside `unlock` we call `safeTransfer`?

For MintTreasury:
- Why inside `burn` we send to `unlock` function?

Where can I find manual/doc for the Treasury DApp?

File: endpoint/contracts/examples/treasury/MintTreasury.sol (hunk 1 above)
Comment (GalaxySciTech) [reply endpoint/contracts/examples/treasury/MintTreasury.sol]: lockTreasury

File: endpoint/contracts/examples/treasury/LockTreasury.sol (hunk 2 above)
Comment (GalaxySciTech) [reply endpoint/contracts/examples/treasury/LockTreasury.sol:39]: mintTreasury

File: endpoint/contracts/examples/treasury/LockTreasury.sol (hunk 3 above)
Comment (GalaxySciTech) [reply endpoint/contracts/examples/treasury/LockTreasury.sol:56]: this function is unlock, it will release the lock balance

**Pull request merged**